launchpad/src/ongoing_operation.rs

```diff
@@ -0,0 +1,114 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+const MIN_GAS_TO_SAVE_PROGRESS: u64 = 100_000_000;
+
+#[derive(TopDecode, TopEncode, TypeAbi, PartialEq)]
+pub enum OngoingOperationType {
+    None,
+    SelectWinners {
+        seed: H256,
+        seed_index: usize,
+        ticket_position: usize,
+    },
+    RestartConfirmationPeriod {
+        ticket_position: usize,
+    },
+}
+
+pub enum GasOp {
+    Load(OngoingOperationType),
+    Continue,
+    Save,
+    Completed,
+}
+
+pub enum LoopOp {
+    Continue,
+    Save(OngoingOperationType),
+    Break,
+}
+
+impl LoopOp {
+    fn is_break(&self) -> bool {
+        return matches!(self, LoopOp::Break);
+    }
+}
+
+#[elrond_wasm::module]
+pub trait OngoingOperationModule {
+    fn run_while_it_has_gas<Process>(
+        &self,
+        initial_op: OngoingOperationType,
```

Comment:
`initial_op` can probably be moved inside this method and be loaded / stored there.